### PR TITLE
Removing numpy dev testing as np 1.18 will be released after the EOS 2.0.x

### PR DESCRIPTION
**Pull request by bsipocz**
As discussed in #9075. 

This PR also limits the numpy version for the docs build as suggested in #9088

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -114,38 +114,14 @@ matrix:
 
         # Try on Windows
         - os: windows
-          env: PYTHON_VERSION=3.7 SETUP_CMD='test --readonly'
```
Comment (pllim) [inline .travis.yml]: Removing `--readonly` seems unrelated. Is there a need for this?

**Comment by pllim** [comment]
p.s. Is the failure of `ci/circleci: image-tests-mpl153` build expected?

File: .travis.yml (hunk 1 above)
Comment (bsipocz) [reply .travis.yml]: yes, `--readonly` was a feature in v3.1, thus we cannot use here. I could have push it directly to the branch or patch the backports as I usually do, but given there was this PR, I would just added to it.

**Comment by bsipocz** [comment]
@pllim - The mpl153 build is failing on the LTS branch already. I go ahead and merge this PR, but that build should be straighten out eventually. I'm hopeful  @astrofrog has or will have a solution.

File: .travis.yml (hunk 1 above)
Comment (pllim) [reply .travis.yml]: Ah, gotcha. Thanks!
